src/redux/auth/reducer.ts

```diff
@@ -0,0 +1,24 @@
+import { AuthActionTypes, AuthState } from '@interfaces/auth';
+
+import { actions } from './actions';
+
+const initalState: AuthState = {
+  user: null,
+  userLoading: false,
+  userError: null
+};
+
+const authReducer = (state = initalState, action: AuthActionTypes): AuthState => {
+  switch (action.type) {
+    case actions.LOGIN:
+      return { ...state, userLoading: true };
+    case actions.LOGIN_SUCCESS:
+      return { ...state, user: action.payload, userLoading: false };
```

Comment:
`return { ...state, user: action.payload, userLoading: false, userError: null };` to delete old errors

Reply:
I love this comment